### DISCUSSION TITLE
How much revlog is enough?

### DISCUSSION THREAD

**Discussion by RickCarlino** [open-spaced-repetition/free-spaced-repetition-scheduler#19] [Q&A] [answered]
I have never used the optimizer. How soon is too soon to use it? Is there a minimum revlog size before it is advisable or useful to use the optimizer? Put differently, is there a penalty to having too small of a revlog when optimizing?

**Comment by L-M-Sherlock** [comment, marked as answer]
TL;DR
1. If the revlog size is less than 8, use default parameters
2. If the size is greater than 8, and less than 64, use pretrained parameters
3. If the size  is greater than 64, use full-optimized parameters
4. We have L1 regularization penalty to prevent the parameters from deviating from the default parameters too far when the revlog size is small.



For details, please see: 

https://github.com/open-spaced-repetition/fsrs-rs/pull/172

https://github.com/open-spaced-repetition/srs-benchmark/blob/main/notebook/minimum_limit.ipynb
